### PR TITLE
First task fourth part

### PR DESCRIPTION
**Pull request by ahmedbilal**

### CODE REVIEW

File: weatherman.py [hunk 1]
```diff
@@ -146,6 +146,22 @@ def avg_mean_humid(self, by = None):
...
 
+
+    def draw_chart_for_month(self, by = None):
+        date = shortToLongDate(by)
+        year = date[0]
+        month = date[1]
+        for e in wmentries.entries.keys():
+            if by is not None and (year not in e or month not in e):
+                continue
+            for d in wmentries.entries[e]:
```
Comment (ahsan-ul-haq) [inline weatherman.py]: avoid using variables like **e, d, by try** to use something meaningful

Comment (ahsan-ul-haq) [reply weatherman.py]: I am guessing you are using dict here so better way to traverse dict in python is something like 
```
for key, value in wmentries.entries.iteritems()
```
and look into iteritems() as well.

File: weatherman.py [hunk 2]
```diff
@@ -146,6 +146,22 @@ def avg_mean_humid(self, by = None):
         else:
             return None
 
+
+    def draw_chart_for_month(self, by = None):
```
Comment (ahsan-ul-haq) [inline weatherman.py]: Read through https://www.python.org/dev/peps/pep-0008/? 
This is code quality standards that we follow. According to this every method should have docstring and here you should not put any spaces around = when use in method. It should be something like
```
def draw_chart_for_month(self, by=None):
```

Comment (ahsan-ul-haq) [reply weatherman.py]: Also class should be inherited from object class of python like following 
```
class WeathermanEntries(object):
```

File: weatherman.py [hunk 3]
```diff
@@ -146,6 +146,22 @@ def avg_mean_humid(self, by = None):
...
+        date = shortToLongDate(by)
+        year = date[0]
+        month = date[1]
+        for e in wmentries.entries.keys():
+            if by is not None and (year not in e or month not in e):
```
Comment (ahsan-ul-haq) [inline weatherman.py]: it can be simplified as 
```
if by and (year not in e or month not in e):
```

File: weatherman.py [hunk 4]
```diff
@@ -146,6 +146,22 @@ def avg_mean_humid(self, by = None):
...
+                continue
+            for d in wmentries.entries[e]:
+                max_temp = d.get('Max TemperatureC')
+                min_temp = d.get('Min TemperatureC')
+                if max_temp != '':
```
Comment (ahsan-ul-haq) [inline weatherman.py]: this can be
```
if max_temp:
```

File: weatherman.py [hunk 5]
```diff
@@ -146,6 +146,22 @@ def avg_mean_humid(self, by = None):
...
+        month = date[1]
+        for e in wmentries.entries.keys():
+            if by is not None and (year not in e or month not in e):
+                continue
+            for d in wmentries.entries[e]:
+                max_temp = d.get('Max TemperatureC')
+                min_temp = d.get('Min TemperatureC')
+                if max_temp != '': 
+                    pkt = d.get("PKT") or d.get("PKST")
+                    print("%10s" % pkt,  sty.fg.blue, "+" * int(min_temp) + sty.fg.rs + sty.fg.red + "+" * int(max_temp), sty.fg.rs, min_temp + "C - " + max_temp + "C")
```
Comment (ahsan-ul-haq) [inline weatherman.py]: try to make sure no line is longer than 120 characters.

File: weatherman.py [hunk 6]
```diff
@@ -171,35 +187,59 @@ def read_file(file):
     meta = f.readline().strip("\n").replace(", ", ",").split(",")
     entries = []
     for line in f.readlines():
+        if "<!--" in line:
+            continue
```
Comment (ahsan-ul-haq) [inline weatherman.py]: What is the purpose of this statement?

File: weatherman.py [hunk 7]
```diff
@@ -171,35 +187,59 @@ def read_file(file):
...
         entries.append(read_to_weatherman_entry(meta, l))
     wmentries.set_entries(file, entries)
 
+def part_one(year):
+    highest_temp = wmentries.highest_temp(year)
+    lowest_temp = wmentries.lowest_temp(year)
+    most_humid = wmentries.most_humid(year)
+    print("Highest:", highest_temp[0], "on", highest_temp[1])
+    print("Lowest:", lowest_temp[0], "on", lowest_temp[1])
+    print("Humidity:", most_humid[0], "on", most_humid[1])
```
Comment (ahsan-ul-haq) [inline weatherman.py]: Read through python string format and try using that.

File: weatherman.py [hunk 8]
```diff
@@ -171,35 +187,59 @@ def read_file(file):
...
+
+
+def part_two(date):
+    highest_avg_temp = wmentries.highest_avg_temp(date)
+    lowest_avg_temp = wmentries.lowest_avg_temp(date)
+    avg_mean_humid = wmentries.avg_mean_humid(date)
+    print("Highest Average:", highest_avg_temp)
+    print("Lowest Average:", lowest_avg_temp)
+    print("Average Mean Humidity:", avg_mean_humid)
```
Comment (ahsan-ul-haq) [inline weatherman.py]: same as above.

File: weatherman.py [hunk 9]
```diff
@@ -171,35 +187,59 @@ def read_file(file):
...
-        print("Highest Average:", highest_avg_temp)
-        print("Lowest Average:", lowest_avg_temp)
-        print("Average Mean Humidity:", avg_mean_humid)
-
+    for i in range(2, len(sys.argv) - 1, 2):
+        mode = sys.argv[i]
+        if mode == '-e':
+            year = sys.argv[i + 1]
+            part_one(year)
+            print()
```
Comment (ahsan-ul-haq) [inline weatherman.py]: why you are adding print()

**Review by ahsan-ul-haq — CHANGES_REQUESTED** [review]
Read python pep8 conventions for better code quality and docstring according to that.

File: weatherman.py (hunk 1 above)
Comment (ahmedbilal) [reply weatherman.py]: ok

**Comment by ahmedbilal** [comment]
@ahsan-ul-haq Please check now

File: weatherman.py [hunk 10]
```diff
@@ -1,155 +1,166 @@
...
+    """ Takes date (as str) in "Year/Month" format
+        and returns date (as tuple) in (year, month_abbrivation_in_3_letters)
+     """
     months = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
-    month = months[int(month) - 1]
-    return (year, month)
 
-class WeathermanEntries:
+    date = arg.split("/")
+    year, month = date[0], months[int(date[1]) - 1]
```
Comment (ahsan-ul-haq) [inline weatherman.py]: Try using python datetime package for such manipulations.

**Review by ahsan-ul-haq — CHANGES_REQUESTED** [review]
One small change

**Review by ahsan-ul-haq — APPROVED** [review]

**Pull request merged**